### PR TITLE
feature/winbindd-refactor

### LINKED ISSUE #3950: packetfence-winbindd: infinite chroot command
On PF 8.2.1 or PF 8.3 running on CentOS 7.6, I see following messages in `/var/log/messages`:

```
Jan 22 17:43:21 pf systemd: Started Session c17965440 of user root.
Jan 22 17:43:21 pf winbindd[12428]: [2019/01/22 16:43:21.858193,  0] ../lib/util/pidfile.c:197(pidfile_create)
Jan 22 17:43:21 pf winbindd[12428]:  ERROR: winbindd is already running. File /usr/local/pf/var/run/domain/winbindd.pid exists and process id 15493 is running.
```

in `journalctl` of `packetfence-winbindd`:
```
Jan 22 17:44:38 pf.domain.local sudo[17501]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/sbin/chroot /chroots/domain /usr/sbin/winbindd -s /etc/samba/domain.conf -l /var/log/sambadomain --foreground
```

These messages appear each second and congest disk space on servers. 

Samba packages have been upgraded in CentOS 7.6, I suspect that PF's code doesn't take care of a change to manage processes.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Refactor to use the fork worker pattern in the other perl daemon

Impacts
=======
winbindd-wrapper

Issue
=====
fixes #3950

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

File: sbin/winbindd-wrapper [hunk 1]
```diff
@@ -202,20 +208,21 @@ sub run_winbindd {
     my $log_directory = "/var/log/samba$id";
 
     my $pid = fork();
-    if($pid) {
+    if (!defined $pid) {
+        #we should just die at this point in time
+        #need to discuss the proper thing to do
```
Comment (julsemaan) [inline sbin/winbindd-wrapper]: Lets discuss!

**Comment by jrouzierinverse** [comment]
Updated

**Comment by julsemaan** [comment]
For me using this PR, none of the winbindd processes start for all of my 3 domains.

Using devel it attempts to start the processes as it should

**Comment by jrouzierinverse** [comment]
Re-test please

**Comment by jrouzierinverse** [comment]
BUmp!

**Comment by jrouzierinverse** [comment]
@julsemaan were you able to retest?

**Comment by julsemaan** [comment]
It seems its starting the processes now but when stopping/restarting it doesn't kill the winbindd processes resulting in stale processes being left there

**Pull request merged**